databricks/koalas/frame.py

```diff
@@ -3557,20 +3572,29 @@ def pivot_table(self, values=None, index=None, columns=None,
         elif isinstance(index, list):
             sdf = self._sdf.groupBy(index).pivot(pivot_col=columns).agg(*agg_cols)
         else:
-            raise ValueError("index should be a None or a list of columns.")
+            raise ValueError("Index should be a None or a list of columns.")
 
         if fill_value is not None and isinstance(fill_value, (int, float)):
             sdf = sdf.fillna(fill_value)
 
         if index is not None:
-            return DataFrame(sdf).set_index(index)
-        else:
             if isinstance(values, list):
-                index_values = values[-1]
+                if len(values) == 2:
+                    data_columns = [column for column in sdf.columns if column not in index]
+                    data_columns.sort(key=lambda x: x.split('_')[1])
```

Comment:
I think this won't work when the function has `_` in its name. can we do `x.split('_', 1)[1]`

Reply:
Thanks for the trick!